src/main/java/de/rabea/player/UnbeatableComputerPlayer.java

```diff
@@ -0,0 +1,81 @@
+package de.rabea.player;
+
+import de.rabea.game.Board;
+import de.rabea.game.Mark;
+import de.rabea.game.Player;
+
+import java.util.List;
+
+public class UnbeatableComputerPlayer extends Player {
+
+    public UnbeatableComputerPlayer(Mark mark) {
+        super(mark);
+    }
+
+    @Override
+    public int getPosition(Board board) {
+        int remainingMovesCount = board.emptyCells().size();
+        return minimax(remainingMovesCount, board, mark).getMove();
+    }
+
+    private ScoredMove minimax(int remainingMovesCount, Board currentBoard, Mark currentMark) {
+        int BEST_MOVE_PLACEHOLDER = -1;
+
+        int bestScore = resetBestScore(currentMark);
+        int bestMove = BEST_MOVE_PLACEHOLDER;
+
+        List<Integer> freePositions = currentBoard.emptyCells();
```

Comment:
You don't need `freePositions` yet. It can even be inlined into the loop declaration.